x/babylon/keeper/wasm.go

```diff
@@ -83,17 +87,58 @@ func (k Keeper) SendEndBlockMsg(c context.Context) error {
 		},
 	}
 
-	// send the sudo call
-	return k.doSudoCall(ctx, finalityAddr, msg)
+	// send the sudo call with gas limits
+	if err := k.doSudoCallWithGasLimit(ctx, finalityAddr, msg); err != nil {
+		k.Logger(ctx).Error("Failed to send EndBlock message to BTC finality contract", "error", err)
+		return fmt.Errorf("BTC finality contract EndBlock call failed: %w", err)
+	}
```

Comment:
This `doSudoCallWithGasLimit` seems less relevant to this PR. It imposes a gas limit over the sudo call. What is the rationale behind it and is there reference to this code?